src/DSharp.Compiler/Generator/CodeGenerator.cs

```diff
@@ -52,6 +52,10 @@ internal static class CodeGenerator
                 foreach (Statement statement in implementation.Statements)
                     StatementGenerator.GenerateStatement(generator, symbol, statement);
             }
+            catch(Exception e)
```

Comment:
maybe we can make the StatementGenerator an IDisposable and call `EndImplementation()` on Dispose

we would use by doing "using....." and then if an exception gets thrown it properly disposes

Reply:
Agreed, I probably won't do that work now. That catch is to ensure we actually do show errors to the build as it kept hiding that fact the script output was blowing up.